client/src/components/frontend/ProjectActivityUpdate.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component, PropTypes } from 'react';
+import {Link} from 'react-router';
+
+export default class ProjectActivityUpdate extends Component {
+
+  static propTypes = {
+    update: PropTypes.object,
+  };
+
+  state = {
+    containerPadding: {
+      paddingTop: '0px',
+      paddingBottom: '0px'
+    }
+  };
+
+  componentDidUpdate = () => {
+    const el = this.refs.activityContainer;
+    const centerPadding = (el.parentNode.offsetHeight - el.offsetHeight) / 2;
+    if (centerPadding > 0) {
+      this.setState({containerPadding: {
+        paddingTop: centerPadding + 'px',
+        paddingBottom: centerPadding + 'px'
+      }})
+    }
+  };
+
+  renderUpdateContent = (update)=> {
+    switch (update.type) {
```

Comment:
Instead of a big switch statement, I'd like to see a switch that calls out to individual functions. Eg, this switch could call renderUpdateComment, renderUpdateFile, etc.

I'm also not crazy about thinking of each boxes as "updates." That word is too fuzzy. Each individual box will represent a project event—a thing that happened in relation to the project. "Updates" is too specific, as this activity can be something external to the project, eg, a tweet with a specific hash tag, rather than an update to the project itself.

I'd call the component EventGrid or EventList. It can be passed any collection of event and render them. If we later decide that texts can have events too, then we can easily repurpose the component for that purpose. Cool?